src/query/functions/fetch.go

```diff
@@ -88,13 +87,16 @@ func (o FetchOp) Node(controller *transform.Controller, storage storage.Storage,
 }
 
 // Execute runs the fetch node operation
-func (n *FetchNode) Execute(ctx context.Context) error {
+func (n *FetchNode) Execute(queryCtx *models.QueryContext) error {
 	timeSpec := n.timespec
 	// No need to adjust start and ends since physical plan already considers the offset, range
 	startTime := timeSpec.Start
 	endTime := timeSpec.End
 	opts := storage.NewFetchOptions()
 	opts.BlockType = n.blockType
+
+	ctx := queryCtx.Ctx
```

Comment:
Would it be worth refactoring `storage` to use the new context?

Reply:
By that do you mean have `storage` take in `queryCtx` instead of `context.Context`? I'd say no; `queryCtx` at least right now is pretty specific to node processing, so I'm inclined to keep its reach limited.